client/lib/reader-tags/tags.js

```diff
@@ -18,10 +20,12 @@ function receiveTags( newTags ) {
 	forOwn( newTags, ( sub ) => {
 		sub.URL = '/tag/' + sub.slug + '/';
 		sub.title = decodeEntities( sub.title );
-		sub.slug = sub.slug.toLowerCase();
+		sub.slug = (sub.slug) ? sub.slug.toLowerCase() : '';
 		tags[ sub.slug ] = sub;
 	} );
-	TagStore.emit( 'change' );
+	if (!(isArray(newTags) && newTags[0].hasOwnProperty('error_data') && newTags[0].error_data.hasOwnProperty('invalid_tag'))) {
```

Comment:
I noticed a issue that when I emitted the change, it would infinitely loop back to the function 'receiveTags', if the tag does not exist. Of course this really stems from a issue with the below code

```javascript
TagStore.dispatchToken = dispatcher.register( function( payload ) {
	const action = payload.action;
	if ( ! action || action.error ) {
		return;
	}
```

where the variable 'payload' clearly has a error, in that the tag is not found, but the property payload.action.error has no error. Part of the issue is that I am not sure what consists of a error. So I went into the file 'client/lib/reader-tags/actions.js' which has the function 'fetchTag' for fetching tags. The function is only looking for server errors and not if there was a error finding something. I am going to test adding something like this into that function

```javascript
error = data.tag.error_data;
```

Let me know if this works?